### PR TITLE
Logistration enable footer and custom form

### PR DESCRIPTION
**Pull request by caseylitton**
@stvstnfrd @caesar2164 PR for enabling logistration footer and customizing logistration registration form with research statement and splitting check boxes for honor code and TOS.   Just remembered, I did not modify the check box for the TOS to also contain the privacy policy.

 enabling logistration footer was just commenting out a line, The conditional at the bottom of the login_register.html would always run no make what I put after the "% if".  So if ENABLE_COMBINED_LOGIN_REGISTRATION is true then the footer is overridden because this html was used.  Maybe syntax?

I added the FOOTER_HTML to devstack.py because I didn't know where else to put it.  It was only in lms.env.json on my box

### CODE REVIEW

File: lms/djangoapps/student_account/views.py [hunk 1]
```diff
@@ -96,14 +96,24 @@ def login_and_registration_form(request, initial_mode="login"):
         except (KeyError, ValueError, IndexError):
             pass
 
+
```
Comment (caesar2164) [inline lms/djangoapps/student_account/views.py]: remove extra line

File: lms/envs/devstack.py [hunk 2]
```diff
@@ -230,3 +230,5 @@ def should_show_debug_toolbar(_):
...
...Stanford's instructional offerings as well as the quality of learning and related sciences worldwide. In the interest of research, you may be exposed to some variations in the course materials. Stanford does not use learner data for any purpose beyond the University's stated missions of education and research. For purposes of research, we may share information we collect from online learning activities, including {pii_link_start}Personally Identifiable Information{link_end}, with researchers beyond Stanford. However, your Personally Identifiable Information will only be shared as permitted by applicable law, will be limited to what is necessary to perform the research, and will be subject to an agreement to protect the data. We may also share with the public or third parties aggregated information that does not personally identify you. Similarly, any research findings will be reported at the aggregate level and will not expose your personal identity.  {paragraph_start}Please read our {privacy_link_start}Privacy Policy{link_end} for more information regarding the processing, transmission and use of data collected through the Stanford Online website and Stanford's Lagunita platform."
```
Comment (caesar2164) [inline lms/envs/devstack.py]: these should be removed from this and PRed to config

File: lms/djangoapps/student_account/views.py (hunk 1 above)
Comment (caesar2164) [inline lms/djangoapps/student_account/views.py]: this will fix a pep8 error you're getting as well...

Comment (caseylitton) [inline lms/djangoapps/student_account/views.py]: Did you see the one about translating a non literal string?  line 101 of lms/djangoapps/student_account/views.py

File: openedx/core/djangoapps/user_api/views.py [hunk 3]
```diff
@@ -669,26 +669,58 @@ def _add_terms_of_service_field(self, form_desc, required=True):
         terms_text = _(u"Terms of Service")
         terms_link = u"<a href=\"{url}\">{terms_text}</a>".format(
             url=marketing_link("TOS"),
-            terms_text=terms_text
+            terms_text=terms_text,
```
Comment (caesar2164) [inline openedx/core/djangoapps/user_api/views.py]: remove this diff

File: openedx/core/djangoapps/user_api/views.py [hunk 4]
```diff
@@ -669,26 +669,58 @@ def _add_terms_of_service_field(self, form_desc, required=True):
...
-        # Translators: "Terms of service" is a legal document users must agree to
-        # in order to register a new account.
-        label = _(
-            u"I agree to the {platform_name} {terms_of_service}."
-        ).format(
-            platform_name=settings.PLATFORM_NAME,
-            terms_of_service=terms_link
-        )
+        # Add privacy policy to terms of service checkbox label and error
+        if settings.REGISTRATION_ADD_PRIVACY_TO_TOS_CHECKBOX:
```
Comment (caesar2164) [inline openedx/core/djangoapps/user_api/views.py]: I don't think we need to add this conditional for this. All of our instance use the same two link set up for the ToS checkbox or replace the whole login page like CME will do

File: lms/envs/common.py [hunk 5]
```diff
@@ -2235,6 +2235,8 @@
     'city': 'hidden',
     'country': 'hidden',
 }
+REGISTRATION_FOOTER_HTML = ''
```
Comment (stvstnfrd) [inline lms/envs/common.py]: We should give this a more descriptive name, especially since we have an unrelated change in the same PR that actually does show/hide the global footer on the registration page.

Maybe something like `REGISTRATION_FORM_DISCLAIMER_HTML`?
I'm open to suggestions, but we should opt for something more specific than `REGISTRATION_FOOTER_HTML`.

File: lms/static/js/student_account/views/AccessView.js [hunk 6]
```diff
@@ -161,7 +162,8 @@
                         fields: data.fields,
                         model: model,
                         thirdPartyAuth: this.thirdPartyAuth,
-                        platformName: this.platformName
+                        platformName: this.platformName,
+                        registrationFooter: this.registrationFooter,
```
Comment (stvstnfrd) [inline lms/static/js/student_account/views/AccessView.js]: This will probably turn into a merge conflict if edX ever updates this collection.

Specifically, because of the extra added comma after `platformName`, git thinks we're changing more than we are. So if edX ever adds another element to the end of the collection, it will also look like they're changing more too. And if git ever suspects both "sides" modified a line at the same time, that's where conflicts arise.

We can avoid this by adding our new line, `registrationFooter`, basically anywhere but at the very end; we could add it as the first element in the dictionary. Then you can remove the added trailing comma from the `platformName` line.

File: lms/static/js/student_account/views/RegisterView.js [hunk 7]
```diff
@@ -47,7 +48,8 @@
                         errorMessage: this.errorMessage,
                         providers: this.providers,
                         hasSecondaryProviders: this.hasSecondaryProviders,
-                        platformName: this.platformName
+                        platformName: this.platformName,
+                        registrationFooter: this.registrationFooter,
```
Comment (stvstnfrd) [inline lms/static/js/student_account/views/RegisterView.js]: See note above about moving up this insertion.

File: lms/templates/student_account/login_and_register.html [hunk 8]
```diff
@@ -32,5 +32,5 @@
     ## This overwrites the "footer" block declared in main.html
     ## with an empty block, effectively hiding the footer
     ## from logistration pages.
-    <%block name="footer"/>
+    ## <%block name="footer"/>
```
Comment (stvstnfrd) [inline lms/templates/student_account/login_and_register.html]: Don't just comment this out; we should delete it entirely.

Comment (caesar2164) [inline lms/templates/student_account/login_and_register.html]: yeah, just delete lines 31-36

File: openedx/core/djangoapps/user_api/views.py [hunk 9]
```diff
@@ -669,26 +669,58 @@ def _add_terms_of_service_field(self, form_desc, required=True):
         terms_text = _(u"Terms of Service")
         terms_link = u"<a href=\"{url}\">{terms_text}</a>".format(
             url=marketing_link("TOS"),
-            terms_text=terms_text
+            terms_text=terms_text,
         )
 
-        # Translators: "Terms of service" is a legal document users must agree to
```
Comment (stvstnfrd) [inline openedx/core/djangoapps/user_api/views.py]: Wait, I thought you and @caesar2164 had said there was a simple feature flag to support this;
was I mistaken?

Comment (caesar2164) [inline openedx/core/djangoapps/user_api/views.py]: ??

Comment (caseylitton) [inline openedx/core/djangoapps/user_api/views.py]: @stvstnfrd Making the TOS required in REGISTRATION_EXTRAS automatically splits the honor code and tos.  There is not another flag I am aware of that we can use for this task.

File: openedx/core/djangoapps/user_api/views.py (hunk 3 above)
Comment (stvstnfrd) [inline openedx/core/djangoapps/user_api/views.py]: > remove this diff

To clarify, @caseylitton , every line which we commit we do so explicitly. In this case, this change is:
- unnecessary
  - Sure, it's the preferred style (trailing comma), but it doesn't actually add anything substantive in this context.
- unrelated
  - There are times where it makes sense to clean up old syntax, but that goal is outside the scope of this pull request; PRs should be focussed.
- divergent
  - It's one more difference between our code and upstream. Not only does it add noise when comparing origin and upstream, but more importantly, it increases the odds of hitting Yet Another Merge Conflict.

File: openedx/core/djangoapps/user_api/views.py (hunk 9 above)
Comment (caseylitton) [inline openedx/core/djangoapps/user_api/views.py]: @stvstnfrd @caesar2164 There is a paragraph comment in opened/djangoapps/user_api/views.py  line 287 discussing TOS, honor code, and privacy policy check boxes...supporting flexibility..etc.  I know the logistration has been updated in dogwood, so our change may never make it upstream.

This may have been up

Comment (caesar2164) [inline openedx/core/djangoapps/user_api/views.py]: @caseylitton none of the changes in this PR are meant to go upstream

File: lms/static/js/student_account/views/AccessView.js (hunk 6 above)
Comment (caesar2164) [inline lms/static/js/student_account/views/AccessView.js]: I would move our new line to 163 or 164, and remove the added comma on the platformName one

File: lms/static/js/student_account/views/RegisterView.js (hunk 7 above)
Comment (caesar2164) [inline lms/static/js/student_account/views/RegisterView.js]: same here, move new line to 49 or 50, and remove extra comma

File: openedx/core/djangoapps/user_api/views.py (hunk 9 above)
Comment (caesar2164) [inline openedx/core/djangoapps/user_api/views.py]: @stvstnfrd see https://github.com/Stanford-Online/edx-platform/pull/481/files#diff-224fd193a6c7eba663b3764afec7110dR614

File: openedx/core/djangoapps/user_api/views.py (hunk 4 above)
Comment (caesar2164) [inline openedx/core/djangoapps/user_api/views.py]: see changes I'm suggesting in my pr to this branch: https://github.com/caseylitton/edx-platform/pull/1/files

(don't bother merging my PR, just make the changes in your branch)

File: lms/envs/common.py (hunk 5 above)
Comment (caesar2164) [inline lms/envs/common.py]: that seems fine to me

File: lms/envs/aws.py [hunk 10]
```diff
@@ -767,3 +767,11 @@
 
 PROCTORING_BACKEND_PROVIDER = AUTH_TOKENS.get("PROCTORING_BACKEND_PROVIDER", PROCTORING_BACKEND_PROVIDER)
 PROCTORING_SETTINGS = ENV_TOKENS.get("PROCTORING_SETTINGS", PROCTORING_SETTINGS)
+
+REGISTRATION_FOOTER_DISCLAIMER = ENV_TOKENS.get(
+    'REGISTRATION_FOOTER_DISCLAIMER',
+    REGISTRATION_FOOTER_DISCLAIMER
+)
+REGISTRATION_ADD_PRIVACY_TO_TOS_CHECKBOX = ENV_TOKENS.get(
```
Comment (stvstnfrd) [inline lms/envs/aws.py]: Remove

File: lms/envs/common.py [hunk 11]
```diff
@@ -2235,6 +2235,45 @@
...
+        "identify you. Similarly, any research findings will be reported "
+        "at the aggregate level and will not expose your personal "
+        "identity.  {paragraph_start}Please read our "
+        "{privacy_link_start}Privacy Policy{link_end} for more "
+        "information regarding the processing, transmission and use of "
+        "data collected through the {platform_name_on_tos} platform. "
+    ),
+}
+
+REGISTRATION_ADD_PRIVACY_TO_TOS_CHECKBOX = False
```
Comment (stvstnfrd) [inline lms/envs/common.py]: Remove.

**Comment by caesar2164** [comment]
diff looks good! 👍

**Comment by stvstnfrd** [comment]
+1
